datafusion/expr/src/expr.rs

```diff
@@ -362,7 +362,7 @@ pub enum Expr {
     Placeholder(Placeholder),
     /// A placeholder which holds a reference to a qualified field
     /// in the outer query, used for correlated sub queries.
-    OuterReferenceColumn(DataType, Column),
+    OuterReferenceColumn(Box<(DataType, Column)>),
```

Comment:
If we are going to make this change anyways, can we also pull this into a named struct rather than a unnamed tuple

like

```rust
struct OuterReference {
 // fields here
}

enum Expr {
... 
    OuterReferenceColumn(OuterReference),
...
}
```

Reply:
Thank you @alamb for this good suggestion, addressed in latest PR.